File: TypeScript/Practice/LeetCode/Medium/Maximum Number of Coins You Can Get.ts
// https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/

function maxCoins(piles: number[]): number {
    let sum: number = 0;
    let counter: number = 0;
    let idx: number = 1;

    piles.sort((a, b) => b - a);

    while (counter < piles.length / 3) {
        sum += piles[idx];
        idx += 2;
        counter++;
    }

    return sum;
};

File: C#/Practice/LeetCode/Medium/Maximum Number of Coins You Can Get.cs
// https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/

public class Solution {
    public int MaxCoins(int[] piles) {
        int sum = 0;
        int counter = 0;
        int idx = 1;

        Array.Sort(piles, (a, b) => b - a);

        while (counter < piles.Length / 3)
        {
            sum += piles[idx];
            idx += 2;
            counter++;
        }

        return sum;
    }
}